### PR TITLE
Answer:6 structural directives roles authorization

### PR DESCRIPTION
**Pull request by SephShom**
## Checklist for challenge submission

- [ ] Start your PR message with Answer:${challenge_number}

### CODE REVIEW

File: apps/angular/permissions/src/app/commons/role-auth.directive.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  standalone: true,
+})
+export class RoleAuthDirective implements OnInit {
+  private userRoles!: Role[];
+
+  @Input()
+  set hasRole(roles: Role[]) {
+    if (!roles || !roles.length) {
+      throw new Error('Roles value is empty or missed');
+    }
```
Comment (tomalaforge) [inline apps/angular/permissions/src/app/commons/role-auth.directive.ts:22]: you can use required: true

File: apps/angular/permissions/src/app/commons/role-auth.directive.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private userStore: UserStore,
+  ) {}
+
+  ngOnInit() {
+    this.userStore.user$.subscribe({
```
Comment (tomalaforge) [inline apps/angular/permissions/src/app/commons/role-auth.directive.ts:34]: don't forget to unsuscribe

File: apps/angular/permissions/src/app/commons/admin-auth.directive.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  hasRoleSuperAdmin: boolean = false;
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private userStore: UserStore,
+  ) {}
+
+  ngOnInit() {
+    this.userStore.user$.subscribe({
```
Comment (tomalaforge) [inline apps/angular/permissions/src/app/commons/admin-auth.directive.ts:25]: don't forget to unsuscribe

File: apps/angular/permissions/src/app/commons/role-auth.directive.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        if (
+          this.userRoles.some(
+            (r) => user?.roles.findIndex((e) => e === r) !== -1,
+          )
+        ) {
+          //then has access
+          this.viewContainer.createEmbeddedView(this.templateRef);
+        } else {
+          // no access
+          this.viewContainer.clear();
```
Comment (tomalaforge) [inline apps/angular/permissions/src/app/commons/role-auth.directive.ts:45]: this is great. Nice work

File: apps/angular/permissions/src/app/commons/admin-auth.directive.ts (hunk 3 above)
Comment (SephShom) [reply apps/angular/permissions/src/app/commons/admin-auth.directive.ts]: Thank you, I won't next time. Fixed on next commit

**Pull request closed without merging**